services/web-server/src/servers/createSubscriptionServer.js

```diff
@@ -2,16 +2,48 @@ import { SubscriptionServer } from 'subscriptions-transport-ws';
 import { execute, subscribe } from 'graphql';
 import credentials from './credentials.js';
 import formatError from './formatError.js';
+import scopeUtils from 'taskcluster-lib-scopes';
+import taskcluster from 'taskcluster-client';
 
 export default ({ server, schema, context, path }) =>
   SubscriptionServer.create(
     {
       schema,
       execute,
       subscribe,
-      onConnect(params, socket) {
-        return new Promise(resolve => {
-          credentials()(socket.upgradeReq, {}, resolve);
+      async onConnect(params, socket) {
+
+        return new Promise((resolve, reject) => {
+          credentials()(socket.upgradeReq, {}, async () => {
+            try {
+
+              const token = params.Authorization;
+
+              if (!token) {
+                reject();
```

Comment:
I don't think it would be necessary here, as we should rely on scopes. And for `anonymous` role on public deployments users should be able to subscribe still.

Reply:
If client id's are associated with scopes and roles, then how is anonymous role determined without a client id?

Reply:
good question. To my knowledge UI and clients do not send this header if the request is "public" (requires no authentication). And it is then up to the auth service to decide if credentials were included, and if not assume it is an anonymous role. 
So I think it can happen that this code will not see any tokens at all for public deployments, so we should not reject here

Reply:
https://github.com/taskcluster/taskcluster/blob/ded37fc2628f6c9ac291a6969d5ddec117615df5/services/auth/src/signaturevalidator.js#L159

Okay, maybe it's not as trivial, and maybe we'll have to fetch user scopes first and see if those include anonymous or not, and if not manually expand those with anonymous before we compare

Reply:
ah, actually, if you just load (without auth) https://community-tc.services.mozilla.com/api/auth/v1/scopes/current  you would see that it includes anonymous already, so calling `auth.currentScopes()` should be enough

Reply:
Have added a check, if the token is not available; the authClient initialization will take credentials parameter as null and on using this to call currentScopes will give us the scopes of anonymous role.